File: python/lesson 6/in_order_iterative.py
# https://leetcode.com/problems/binary-tree-inorder-traversal/

class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        list, stack, current = [], [], root
        while current or len(stack) > 0:
            if current:
                stack.append(current)
                current = current.left
            else:
                current = stack.pop()
                list.append(current.val)
                current = current.right
        return list

File: javascript/lesson 6/inOrderIterative.js
// https://leetcode.com/problems/binary-tree-inorder-traversal/

function inorderTraversal(root) {
    const list = [], stack = [];
    let current = root;
    while (current || stack.length > 0) {
        if (current) {
            stack.push(current);
            current = current.left;
        } else {
            current = stack.pop();
            list.push(current.val);
            current = current.right;
        }
    }
    return list;
}
